packages/web/src/hooks/swap/data/use-swap-handler.tsx

```diff
@@ -1095,6 +1094,51 @@ export const useSwapHandler = () => {
       });
   }, []);
 
+  useEffect(() => {
+    // Do not execute unless both tokens A and B are selected
+    if (!tokenA?.symbol || !tokenB?.symbol) {
+      return;
+    }
+
+    // Handle default values when no user input exists
+    if (estimateFlagRef.current === 0) {
+      if (defaultTokenAAmount) {
+        estimateFlagRef.current += 1;
+        changeTokenAAmount(defaultTokenAAmount);
+        return;
+      }
+      if (defaultTokenBAmount) {
+        estimateFlagRef.current += 1;
+        changeTokenBAmount(defaultTokenBAmount);
+        return;
+      }
+    }
+
+    // Special handling for isSameToken case
+    if (isSameToken && tokenAAmount && !tokenBAmount) {
+      setTokenBAmount(tokenAAmount);
+      return;
+    }
+
+    // Check for user input only for non-same token cases
+    if (!isSameToken) {
+      const isUserInput = tokenAAmount !== "" || tokenBAmount !== "";
+      if (isUserInput) {
+        return;
+      }
+    }
```

Comment:
consider reordering conditions. cheking `!isSameToken` first allows removing redundant `isSameToken` checks in another conditions.

```suggestion
    // Check for user input only for non-same token cases
    if (!isSameToken) {
      const isUserInput = tokenAAmount !== "" || tokenBAmount !== "";
      if (isUserInput) {
        return;
      }
    }

    // Special handling for isSameToken case
    if (tokenAAmount && !tokenBAmount) {
      setTokenBAmount(tokenAAmount);
      return;
    }
```

Reply:
Code is now more readable, taking into account condition reordering!